### ISSUE TITLE
Connecting to home-assistant over tor/onion URL

### ISSUE THREAD

**Issue by marvin8** [home-assistant/home-assistant-android#313]
Hi

I have my local home-assistant url forwarded using tor and am able to access the web interface of home-assistant with the tor-browser from anywhere.

Would it be possible to allow the android app to connect to my home-assistant install using the tor .onion address as well?

Cheers
Mark

**Comment by balloob** [comment]
That won't be possible because it requires us to integrate a full tor-network client into our app, which is out of scope.

**Comment by maps2002** [comment]
Hi,

An alternative is to use Orbot and its VPN mode to "proxy" the Home Assistant App thru Orbot. Be careful with the Android "Private DNS" feature. If it is enabled, the Home Assistant App will not resolve the .onion domain.

One side effect is that when you proxy the Home Assistant App with Orbot, when you are on your home network, it will try to resolve the internal domain with Orbot, and of course it does not work. So you have to disable/enable Orbot on your internal network.

I don't know if it makes sense for the App, but two possible improvements could be:
 - Allow Home Assistant App to use a proxy --> This way we would avoid the Orbot VPN mode, and just use its proxy
 - Allow to bypass the proxy when on the home network --> This check, would allow the Home Assistant App, to "just work" independently of the "external" and "internal" networks that you are connected

What do you think of this?
Thank you!

**Comment by CarlosGS** [comment]
@maps2002 @marvin8 Which solution did you end up using for remote HASS access?

**Comment by marvin8** [comment]
@CarlosGS I have set-up my own VPN server on a vps using wireguard and [PiVPN](https://pivpn.io/). I did this for other reasons but when I connected the server I run home-assistant on to this VPN as well as my Android phone, I just tried accessing home-assistant through the VPN ip address and that worked.

I played around a bit and found that this set-up works really well for me no matter what connection my android is on to see the internet / my VPN.

It works well for me over mobile data and surprising also works well when the phone is using WiFi connected to the same local network as my home-assistant server and I don't need to change connection details to home-assistant.

I chose wireguard for my VPN set-up as it is a lot more stable than OpenVPN in my experience.
